### ISSUE TITLE
ddlm_validate: datetime checks are too strict

### ISSUE THREAD

**Issue by vaitkus** [cod-developers/cod-tools#13]
It was reported in a another issue (#12) that datetime check are a bit too strict when compared to RFC3339.

The following strings were reported as incorrectly identified as invalid full-date or date-time productions:
```
1912-02-03T11:47Z
1990-07-13T14:40
2015-10-30T22:45-02:00
```
However, after checking the ABNF grammar in section 5.6 of [RFC3339](https://www.ietf.org/rfc/rfc3339.txt), I am of the opinion that these values are reported correctly in regards to the current datetime definition as the 'date-time' constructions require both the seconds part and the offset to be explicitly given. Adding the placeholder ":00" seconds allows the values to be successfully validated:
```
1912-02-03T11:47:00Z
1990-07-13T14:40:00Z
2015-10-30T22:45:00-02:00
```
Of course, the checks could be updated to allow constructs without seconds or offsets, but I would like to do that only after the DDLm datetime definition is changed accordingly to not introduce any ad hoc exceptions. However, should such non-standard datetimes be allowed at all?

**Comment by jamesrhester** [comment]
I agree with your analysis. However, after fixing the examples as above the final example still does not validate. See [this run](https://github.com/jamesrhester/Powder_Dictionary/actions/runs/3155935589/jobs/5135129617).
